### PR TITLE
[WIP] Inconsistent termination message output

### PR DESCRIPTION
**Pull request by pontiyaraja**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: This PR to resolve the issue [#70990](https://github.com/kubernetes/kubernetes/issues/70990)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: [#70990]([url](https://github.com/kubernetes/kubernetes/issues/70990))
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_container.go [hunk 1]
```diff
@@ -418,6 +418,7 @@ func (m *kubeGenericRuntimeManager) getPodContainerStatuses(uid kubetypes.UID, n
 			}
 			// Use the termination message written by the application is not empty
 			if len(tMessage) != 0 {
+				klog.Info("PONTIYA   ======================   STATUS ======= LOG  :  ", tMessage, "--Here-------------")
```
Comment (justinsb) [inline pkg/kubelet/kuberuntime/kuberuntime_container.go]: Did you mean to leave this message in here?  It looks like debug code?

File: pkg/kubelet/kuberuntime/logs/logs.go [hunk 2]
```diff
@@ -273,7 +276,8 @@ func ReadLogs(ctx context.Context, path, containerID string, opts *LogOptions, r
 		return fmt.Errorf("failed to open log file %q: %v", path, err)
 	}
 	defer f.Close()
-
+	b, _ := ioutil.ReadFile(path)
+	klog.Info("COMPLETE   ===========================   FILE   DATA  =================  ", string(b))
```
Comment (justinsb) [inline pkg/kubelet/kuberuntime/logs/logs.go]: This feels like a very big change - we're logging the full logs?

File: pkg/kubelet/kuberuntime/logs/logs.go [hunk 3]
```diff
@@ -273,7 +276,8 @@ func ReadLogs(ctx context.Context, path, containerID string, opts *LogOptions, r
 		return fmt.Errorf("failed to open log file %q: %v", path, err)
 	}
 	defer f.Close()
-
+	b, _ := ioutil.ReadFile(path)
```
Comment (justinsb) [inline pkg/kubelet/kuberuntime/logs/logs.go]: Should probably log the error

File: test/e2e/common/runtime.go [hunk 4]
```diff
@@ -248,144 +255,109 @@ while true; do sleep 1; done
...
+					}
+					if i < flakeRetry {
+						framework.Logf("No.%d attempt failed: %v, retrying...", i, err)
+					} else {
+						framework.Failf("All %d attempts failed: %v", flakeRetry, err)
+					}
+				}
 			}
+
+			It("should not be able to pull image from invalid registry", func() {
```
Comment (justinsb) [inline test/e2e/common/runtime.go]: Is there a reason for splitting out the grid tests into individual tests?  I think I like it, but I'm wondering if an actual decision was made somewhere that is motivating this...

**Review by justinsb — COMMENTED** [review]
Got half way through and realized this is likely a WIP?

File: test/e2e/common/runtime.go (hunk 4 above)
Comment (pontiyaraja) [reply test/e2e/common/runtime.go]: Yeah, actually I try to convert these to conformance tests. But walker.go isn't reading the test case string from grid test. So splitting out the grid test.

File: pkg/kubelet/kuberuntime/logs/logs.go (hunk 2 above)
Comment (pontiyaraja) [reply pkg/kubelet/kuberuntime/logs/logs.go]: Yeah, only it's failing only in GKE. 
Actually gke is writing the container termination status message with `\n`.
None of the other environment writing like this. Please check kubelet log. Search DONE\n

File: pkg/kubelet/kuberuntime/kuberuntime_container.go (hunk 1 above)
Comment (pontiyaraja) [reply pkg/kubelet/kuberuntime/kuberuntime_container.go]: No, I will remove it.

File: test/e2e/common/runtime.go (hunk 4 above)
Comment (k-toyoda-pi) [reply test/e2e/common/runtime.go]: I found similar PR of runtime.go. [https://github.com/kubernetes/kubernetes/pull/70492](url)
Could you tell me the relation?

Comment (pontiyaraja) [reply test/e2e/common/runtime.go]: @k-toyoda-pi  this issue came from that PR, this PR is working in progress. And this PR to fix only inconsistent message.

Comment (k-toyoda-pi) [reply test/e2e/common/runtime.go]: Thanks for reply. I understand you are investigating about "\n" with the PR.

File: pkg/kubelet/kuberuntime/logs/logs.go [hunk 5]
```diff
@@ -337,12 +341,15 @@ func ReadLogs(ctx context.Context, path, containerID string, opts *LogOptions, r
 				return fmt.Errorf("failed to get parse function: %v", err)
 			}
 		}
+
+		klog.Info(" LLLLLLLLLLLLLLLLLLLLLLL   ===========================    ", string(l))
```
Comment (changyaowei) [inline pkg/kubelet/kuberuntime/logs/logs.go]: I don't think it's a good message format

**Comment by msau42** [comment]
/hold
This is just a test debugging PR and is not meant to be merged

**Comment by pontiyaraja** [comment]
@kubernetes/sig-node-bugs

**Comment by mgdevstack** [comment]
/close
xref https://github.com/kubernetes/kubernetes/issues/70990

**Pull request closed without merging**